libbpf-cargo/src/build.rs

```diff
@@ -135,6 +143,31 @@ fn compile_one(debug: bool, source: &Path, out: &Path, clang: &Path, options: &s
         .arg("-o")
         .arg(out);
 
+    // Ensure the generated object file is deterministic.
+    if let Some(path) = header_dir {
+        let path = path.path().to_str().unwrap();
+        cmd.arg(format!("-I{}", path));
+        cmd.arg(format!(
+            "-ffile-prefix-map={}=/LIBBPF_INCLUDE_HEADERS",
```

Comment:
Where does LIBBPF_INCLUDE_HEADERS get set? is this an environment variable?

Reply:
No, none of these are variables. These are just hard-coded strings that will appear in the output binary instead of the actual path, you can see them in the readelf dump: https://github.com/libbpf/libbpf-rs/pull/169#issuecomment-1004367500

Reply:
Actually, I think remapping the source directory might make remapping $HOME unnecessary, that would let me remove the `dirs2` dependency.

Reply:
Did that :) and still didn't see any references to `/home`.

You might find it useful - this is the script I've been using to test determinism:
```sh
#!/bin/sh
set -eu

cd "$(git rev-parse --show-toplevel)"
cargo install --offline --debug --path libbpf-cargo
cd ~/redacted
cargo libbpf make --clang-path=./my_clang.sh
git add src

for i in `seq 1 3`; do
	# track changes to .o
	cp target/bpf/redacted.bpf.o .

	cargo libbpf build --clang-path=./my_clang.sh
	cmp redacted.bpf.o target/bpf/redacted.bpf.o
	cargo libbpf gen
	git diff --name-only --exit-code src
done

# make sure there's no references to the home directory
readelf -S redacted.bpf.o  | grep -o ' \.[^ ]*' | tr -d ' ' | while read section; do
	echo $section
	! readelf -x $section redacted.bpf.o | grep /home
done
```